File: lib/models/settlement.dart
class Settlement {

  final String fromID;
  final String toID;
  final double amount;

  Settlement({
    required this.fromID,
    required this.toID,
    required this.amount,
  });

  factory Settlement.fromJson(Map<String, dynamic> json) {
    return Settlement(
      fromID: json["fromID"],
      toID: json["toID"],
      amount: (json['amount'] as num).toDouble()
    );
  }

}

File: backend/internal/models/settlement.go
package models

import "go.mongodb.org/mongo-driver/v2/bson"

type Settlement struct {
	FromUser  bson.ObjectID `bson:"fromUser" json:"fromID"`
	ToUser    bson.ObjectID `bson:"toUser" json:"toID"`
	Amount    float64       `bson:"amount" json:"amount"`
}